homeassistant/components/tesla/__init__.py

```diff
@@ -221,6 +222,12 @@ def unique_id(self) -> str:
     @property
     def icon(self):
         """Return the icon of the sensor."""
+        if self.tesla_device.has_battery():
+            return icon_for_battery_level(
```

Comment:
Entities with device class set, shouldn't set a custom icon.

Reply:
I think the issue is the inheritance though. Since everything inherits from this class, theres no way to not have icon property. unless im missing something.

Reply:
We can return `None` in the icon property for those entities that set device class.

Reply:
ah so UI handles it based on device class then?

Reply:
#37526